packages/matter.js/src/cluster/Cluster.ts

```diff
@@ -176,5 +180,6 @@ export const ClusterExtend =
             attributes: Merge(attributes, attributesExtend),
             commands: Merge(commands, commandsExtend),
             events: Merge(events, eventsExtend),
+            extension: true,
```

Comment:
why this makes a difference? Alao normal "Cluster" can be used nor ally. I would really just use if it is exported it is externally available as a cluster variant": irrelevant if CLuster or ClusterExtend

Reply:
I needed this because in DeviceTypes the cluster is only referenced by ID but the ID is shared across the extensions.  So we need to know what is the "required" cluster and what is optional optional extension.  But this does cause other problems, see my comment r.e. pumps offering "OnOffLighting" as an option.

Maybe we could make DeviceTypes reference the cluster definitions directly rather than the cluster ID?

Reply:
This do not work because it just says "it hast cluster XY",  but not neededly "which feature variant of it". In fact even more bad (see device library specs 5.3.5.2 ... 
But in fact the Pump I would more say it is OnOffWithLevelControl, but I would not add the Lighing one. And when we are rellay correctwe also need special OnOff Handlers ... no idea how we can manage this compexity :-(

Reply:
OK, it's only a problem now in the sense that unnecessary features are in the "options" object.  We don't add them by default because we don't know if they're necessary.  So this just puts it on the user.  They presumably know what features they need so I think it's all good.

Reply:
In fact users need to know which featureset they want - for the "device builders"